stubs/Reflection.phpstub

```diff
@@ -14,13 +14,194 @@ class ReflectionClass implements Reflector {
 
     /**
      * @param T|class-string<T>|interface-string<T>|trait-string|enum-string<T> $argument
+     * @psalm-pure
      */
     public function __construct($argument) {}
 
     /**
      * @return class-string<T>
+     * @psalm-pure
      */
     public function getName(): string {}
+    
+    /** @psalm-pure */
+    public function isInternal(): bool {}
+
+    /** @psalm-pure */
+    public function isUserDefined(): bool {}
+
+    /** @psalm-pure */
+    public function isInstantiable(): bool {}
+
+    /** @psalm-pure */
+    public function isCloneable(): bool {}
+
+    /**
+     * @return non-empty-string|false
+     * @psalm-pure
+     */
+    public function getFileName() {}
+
+    /**
+     * @return positive-int|false
+     * @psalm-pure
+     */
+    public function getStartLine() {}
+
+    /**
+     * @return positive-int|false
+     * @psalm-pure
+     */
+    public function getEndLine() {}
+
+    /**
+     * @return non-empty-string|false
+     * @psalm-pure
+     */
+    public function getDocComment() {}
+
+    /** @psalm-pure */
+    public function getConstructor(): ?ReflectionMethod {}
+
+    /** @psalm-pure */
+    public function hasMethod(string $name): bool {}
+
+    /**
+     * @param non-empty-string $name
+     *
+     * @psalm-pure
+     * @throws ReflectionException
+     */
+    public function getMethod(string $name): ReflectionMethod {}
```

Comment:
`non-empty-string` is not right: https://github.com/Roave/BetterReflection/issues/1278

Reply:
This is about preventing nonsensical code though, not a runtime check